zubhub_backend/zubhub/activities/tests/test_utils.py

```diff
@@ -0,0 +1,223 @@
+    inspiring_artist_data = {
+      'name': 'John Doe',
+      'short_biography': 'A short biography about John Doe',
+      'image': image_map
+    }
+    
+    # call the function being tested
+    artist = create_inspiring_artist(inspiring_artist_data)
+    
+    self.assertIsInstance(artist, InspiringArtist)
+    self.assertEqual(artist.name, inspiring_artist_data['name'])
+    self.assertNotEqual(artist.name, 'Jane Doe')
+    self.assertEqual(artist.image, inspiring_artist_data['image'])
+    self.assertTrue(InspiringArtist.objects.filter(name='John Doe').exists())
+
+
+  def test_update_image(self):
+    new_data = {
+      'file_url': 'https://picsum.photos/400', # notice the 400 instead of 300
+      'public_id': self.test_image_data['public_id'],
+    }
+    
+    new_image = update_image(self.test_image, new_data)
+    
+    self.assertIsInstance(new_image, Image)
+    db_query = Image.objects.get(public_id=new_data['public_id'])
+    # assert that the changes have reflected in the database
+    self.assertEqual(db_query.file_url, new_data['file_url'])
+    self.assertNotEqual(new_image.file_url, self.test_image_data['file_url'])
+    
+  def test_create_activity_images(self):
+    images = [
+      {
+        'image': {
+          'file_url': 'https://picsum.photos/500',
+          'public_id': 'Activity image 1',
+        }
+      },
+      {
+        'image': {
+          'file_url': 'https://picsum.photos/600',
+          'public_id': 'Activity image 2',
+        }
+      }
+    ]
+    
+    # get activity images count & images count before function
+    activity_image_count_before = ActivityImage.objects.filter(activity=self.test_activity).count()
+    image_count_before = Image.objects.all().count()
+    # create activity images
+    create_activity_images(activity=self.test_activity, images=images)
+    # get activity images count & images count after function
+    activity_image_count_after = ActivityImage.objects.filter(activity=self.test_activity).count()
+    image_count_after = Image.objects.all().count()
+    # assert that images and activity images were created
+    self.assertEqual(activity_image_count_after - activity_image_count_before, len(images))
+    self.assertEqual(image_count_after - image_count_before, len(images))
+    
+  def test_create_making_steps(self):
+    making_steps = [
+      {
+        'title': 'Step 1',
+        'description': 'Description for step 1',
+        'step_order': 1,
+        'image': [{'file_url': 'https://picsum.photos/710','public_id': '1',}]
+      },
+      {
+        'title': 'Step 2',
+        'description': 'Description for step 2',
+        'step_order': 2,
+        'image': [{'file_url': 'https://picsum.photos/720','public_id': '2',}]
+      },
+      {
+        'title': 'Step 3',
+        'description': 'Description for step 3',
+        'step_order': 3,
+        'image': [{'file_url': 'https://picsum.photos/730','public_id': '3',}]
+      },
+    ]
+    
```

Comment:
When performing tests we need to have the images and other items created as instances of our models at runtime.